tests/scancode/test_cli.py

```diff
@@ -169,6 +169,15 @@ def test_scan_info_does_collect_infos_with_root(monkeypatch):
     assert 'Scanning done' in result.output
     check_scan(test_env.get_test_loc('info/basic.rooted.expected.json'), result_file, regen=False)
 
+def test_scan_can_handle_long_file_names(monkeypatch):
+    monkeypatch.setattr(click._termui_impl, 'isatty', lambda _: True)
+    test_dir = test_env.get_test_loc('long_file_name/0123456789012345678901234567890123456789.c')
+    runner = CliRunner()
+    result_file = test_env.get_temp_file('json')
+    result = runner.invoke(cli.scancode, ['--copyright', test_dir, result_file], catch_exceptions=True)
+    assert result.exit_code == 0
+    assert 'Scanning done' in result.output
+    check_scan(test_env.get_test_loc('long_file_name/expected.json'), result_file, regen=False)
```

Comment:
This is not needed. We do not care about the scan. On the other hand you make no assertions on the shortening of the name in the UI, so you are not testing anything there